repos/system_upgrade/common/actors/scancpu/libraries/scancpu.py

```diff
@@ -1,22 +1,32 @@
+import json
 import re
 
 from leapp.libraries.common.config import architecture
+from leapp.libraries.common.config.version import get_source_major_version
 from leapp.libraries.stdlib import api, CalledProcessError, run
 from leapp.models import CPUInfo, DetectedDeviceOrDriver, DeviceDriverDeprecationData
 
-LSCPU_NAME_VALUE = re.compile(r'(?P<name>[^:]+):\s+(?P<value>.+)\n?')
+LSCPU_NAME_VALUE = re.compile(r'(?P<name>[^:]+):[^\S\n]+(?P<value>.+)\n?')
 PPC64LE_MODEL = re.compile(r'\d+\.\d+ \(pvr (?P<family>[0-9a-fA-F]+) 0*[0-9a-fA-F]+\)')
 
 
-def _get_lscpu_output():
+def _get_lscpu_output(output_json=False):
     try:
-        result = run(['lscpu'])
+        result = run(['lscpu', '-J' if output_json else ''])
         return result.get('stdout', '')
     except (OSError, CalledProcessError):
         api.current_logger().debug('Executing `lscpu` failed', exc_info=True)
     return ''
 
 
+def _parse_lscpu_output():
+    if get_source_major_version() == '7':
+        return dict(LSCPU_NAME_VALUE.findall(_get_lscpu_output()))
+
+    parsed_json = json.loads(_get_lscpu_output(output_json=True))
```

Comment:
iiuc this will throw a JSONDecodeError if lscpu execution fails.

Reply:
Sure! I think if lscpu execution fails we will catch it in ``_get_lscpu_output`` already. But let's add a sanity check on the structure of the json output

Reply:
Thank you! Yeah, if lscpu fails we will log an error and return an empty string from _get_lscpu_output. My concern is that `json.loads('')` is a guaranteed exception.

Reply:
Added! (had to catch ValueError as JSONDecodeError is not in 2.7)